### PR TITLE
Copied analyzers from dotnet/runtime

### PR DESCRIPTION
**Pull request by bozturkMSFT**
Partly delivers https://github.com/dotnet/templating/issues/2767.

This PR the analyzers and the ruleset. All the violated rules are temporarily showing up as "suggestions". We will change them back to warnings once we can refactor the existing code. Refactored files are already available on my local machine, but I will postpone pushing them for now to prevent conflicts with existing PRs.

This should be ready to merge after the reviews.

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -10,7 +10,7 @@
         <Product>Microsoft .NET Core</Product>
         <Copyright>$(CopyrightNetFoundation)</Copyright>
         <PackageLicenseExpression>MIT</PackageLicenseExpression>
-        <NoWarn>NU5105;NU5128</NoWarn>
+        <NoWarn>NU5105;NU5128;0419,0649</NoWarn>
```
Comment (vlada-shubina) [inline Directory.Build.props:13]: Is this change intentional? Why it is needed?

File: src/Microsoft.TemplateEngine.VisualStudioWizard/Microsoft.TemplateEngine.VisualStudioWizard.csproj [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
     <PropertyGroup>
         <TargetFramework>$(NETFullTargetFramework)</TargetFramework>
+        <EnableAnalyzers>true</EnableAnalyzers>
```
Comment (vlada-shubina) [inline src/Microsoft.TemplateEngine.VisualStudioWizard/Microsoft.TemplateEngine.VisualStudioWizard.csproj:4]: This project is not used anymore. Probably we should remove it from repo

**Review by vlada-shubina — COMMENTED** [review]
Should we add analyzers also to test projects?

File: Directory.Build.props (hunk 1 above)
Comment (bozturkMSFT) [reply Directory.Build.props:13]: Copied from dotnet/runtime https://github.com/dotnet/runtime/blob/3d8d29731b62c6bdc37c6fb3e22ef8bcdd5ec99b/src/coreclr/System.Private.CoreLib/System.Private.CoreLib.csproj#L46

These rules are conflicting with other rules that we have in `CodeAnalysis.ruleset` and need to be disabled. Looking at how runtime repo handles the situation, I believe these are (rules that have IDs without any letters at the beginning) old analyzer rules and cannot be deactivated using the ruleset file. Even when they are not disabled, their warnings don't show up in VS, they make no code fix suggestions. You can only see them if you run `dotnet build`. They also break the build since we treat warnings as errors.

File: src/Microsoft.TemplateEngine.VisualStudioWizard/Microsoft.TemplateEngine.VisualStudioWizard.csproj (hunk 2 above)
Comment (bozturkMSFT) [reply src/Microsoft.TemplateEngine.VisualStudioWizard/Microsoft.TemplateEngine.VisualStudioWizard.csproj:4]: I created a new task for it here: https://github.com/dotnet/templating/issues/2796

**Review by DavidKarlas — APPROVED** [review]
I didn't take a very close look at error/info/warning list of rules, but we are going to tweak them over time anyway....

**Review by vlada-shubina — APPROVED** [review]

File: src/Microsoft.TemplateEngine.Abstractions/Microsoft.TemplateEngine.Abstractions.csproj [hunk 3]
```diff
@@ -4,5 +4,6 @@
         <TargetFrameworks Condition="'$(PackSpecific)' == 'true'">$(NETStandardTargetFramework)</TargetFrameworks>
         <Description>Contracts for extending Template Engine</Description>
         <IsPackable>true</IsPackable>
+        <EnableAnalyzers>true</EnableAnalyzers>
```
Comment (MichaelSimons) [inline src/Microsoft.TemplateEngine.Abstractions/Microsoft.TemplateEngine.Abstractions.csproj:7]: I realize this PR has been merged, but this change is impacting source-build.  The changes like this which explicitly set EnableAnalyers, overrides the value specified in eng/Analyzers.props.  This nuance impacts source-build which disables the analyers because they aren't part of source-build.
